pkg/kubelet/network/plugins.go

```diff
@@ -47,6 +48,20 @@ type NetworkPlugin interface {
 
 	// TearDownPod is the method called before a pod's infra container will be deleted
 	TearDownPod(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) error
+
+	// Status is the method called to obtain the ipv4 and pv6 addresses of the container
+	Status(namespace string, name string, podInfraContainerID kubeletTypes.DockerID) (*NetworkStatus, error)
+}
+
+type InterfaceInfo struct {
+	Iface   net.Interface
+	IPAddrs []net.IP
+}
+
+// Stores the network status of a pod (currently just the network addresses and interfaces)
+type NetworkStatus struct {
+	IPAddress         net.IP // Primary ipv4/ipv6 address of the pod
```

Comment:
What does it mean to have a "primary" IP, when you also allow for an arbitrary number of interfaces each of which can have an arbitrary number of IPs?